core/src/main/java/hudson/Util.java

```diff
@@ -1432,36 +1432,76 @@ public static Number tryParseNumber(@CheckForNull String numberStr, @CheckForNul
+        // the unit tests explicitly test for that as working.
+        if (!base.isAssignableFrom(derived)) {
+            throw new IllegalArgumentException("The specified derived class (" + derived.getCanonicalName() + ") does not derive from the specified base class (" + base.getCanonicalName() + ").");
+        }
+        final Method baseMethod = Util.getMethod(base, null, methodName, types);
+        if (baseMethod == null) {
+            throw new IllegalArgumentException("The specified method is not declared by the specified base class (" + base.getCanonicalName() + "), or it is private, static or final.");
+        }
+        final Method derivedMethod = Util.getMethod(derived, base, methodName, types);
+        // the lookup will either return null or the base method when no override has been found (depending on whether
+        // the base is an interface)
+        return derivedMethod != null && derivedMethod != baseMethod;
     }
 
-    private static Method getMethod(@NonNull Class clazz, @NonNull String methodName, @NonNull Class... types) {
-        Method res = null;
+    private static Method getMethod(@NonNull Class<?> clazz, @Nullable Class<?> base, @NonNull String methodName, @NonNull Class<?>... types) {
         try {
-            res = clazz.getDeclaredMethod(methodName, types);
-            // private, static or final methods can not be overridden
-            if (res != null && (Modifier.isPrivate(res.getModifiers()) || Modifier.isFinal(res.getModifiers()) 
-                    || Modifier.isStatic(res.getModifiers()))) {
-                res = null;
+            final Method res = clazz.getDeclaredMethod(methodName, types);
+            final int mod = res.getModifiers();
+            // private and static methods are never ok, and end the search
+            if (Modifier.isPrivate(mod) || Modifier.isStatic(mod)) {
+                return null;
             }
+            // when looking for the base/declaring method, final is not ok
+            if (base == null && Modifier.isFinal(mod)) {
+                return null;
+            }
+            // when looking for the overriding method, abstract is not ok
+            if (base != null && Modifier.isAbstract(mod)) {
+                return null;
+            }
+            return res;
         } catch (NoSuchMethodException e) {
+            // If the base is an interface, the implementation may come from a default implementation on a derived
+            // interface. So look at interfaces too.
+            if (base != null && Modifier.isInterface(base.getModifiers())) {
+                for (Class<?> iface : clazz.getInterfaces()) {
+                    if (base.equals(iface) || !base.isAssignableFrom(iface)) {
+                        continue;
+                    }
+                    final Method defaultImpl = Util.getMethod(iface, base, methodName, types);
```

Comment:
ditto this variable naming

Reply:
No, because an interface can only inherit from other interfaces, and `getMethod` won't return abstract methods, so here it either returns a default-implemented method or `null`.